universal-application-tool-0.0.1/app/views/questiontypes/RepeaterQuestionRenderer.java

```diff
@@ -36,43 +37,70 @@ public RepeaterQuestionRenderer(ApplicantQuestion question) {
+    for (int index = 0; index < entityNames.size(); index++) {
+      enumeratorFields.with(existingEnumeratorField(Optional.of(entityNames.get(index)), index));
+    }
 
     return div()
         .withClasses(Styles.MX_AUTO, Styles.W_MAX)
         .with(
             div()
                 .withClasses(ReferenceClasses.APPLICANT_QUESTION_TEXT)
                 .withText(question.getQuestionText()),
+            div()
+                .withId(PLACEHOLDER_ID)
+                .withClass(Styles.HIDDEN)
+                .withText(enumeratorQuestion.getPlaceholder()),
             div()
                 .withClasses(
                     ReferenceClasses.APPLICANT_QUESTION_HELP_TEXT,
                     Styles.TEXT_BASE,
                     Styles.FONT_THIN,
                     Styles.MB_2)
                 .withText(question.getQuestionHelpText()),
-            repeaterFields,
-            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addRepeaterEntity")))
-        .withType("button");
+            enumeratorFields,
+            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addEnumeratorEntity"))
+                .withType("button"),
+            fieldErrors(enumeratorQuestion.getQuestionErrors()));
   }
 
-  public static Tag repeaterField(
-      String placeholderText, Optional<String> existingOption, int index) {
+  /**
+   * Crete an enumerator field for existing entries. These come with a checkbox to delete during
+   * form submission.
+   */
+  private Tag existingEnumeratorField(Optional<String> existingOption, int index) {
     ContainerTag optionInput =
         FieldWithLabel.input()
-            .setFieldName(RepeaterQuestionDefinition.REPEATED_ENTITY_NAME_KEY + "[]")
-            .setPlaceholderText(placeholderText)
+            .setFieldName(question.getContextualizedPath().toString())
             .setValue(existingOption)
             .getContainer()
             .withClasses(Styles.FLEX, Styles.ML_2);
     Tag removeOptionBox =
         FieldWithLabel.checkbox()
-            .setFieldName(RepeaterQuestionDefinition.REPEATED_ENTITY_NAME_KEY + "_delete[]")
+            .setFieldName(Path.empty().join(Scalar.DELETE_ENTITY).asArrayElement().toString())
             .setValue(String.valueOf(index))
             .getContainer();
 
-    return div().withClasses(REPEATER_FIELD_CLASSES).with(optionInput, removeOptionBox);
+    return div().withClasses(ENUMERATOR_FIELD_CLASSES).with(optionInput, removeOptionBox);
+  }
+
+  /**
+   * Create an enumerator field template for new entries. These come with a button to delete itself.
+   */
+  public static Tag newEnumeratorFieldTemplate(Path contextualizedPath) {
+    ContainerTag optionInput =
+        FieldWithLabel.input()
+            .setFieldName(contextualizedPath.toString())
+            .getContainer()
+            .withClasses(Styles.FLEX, Styles.ML_2);
+    Tag removeFieldButton = button("x").withType("button").withClasses(Styles.FLEX, Styles.ML_4);
```

Comment:
For a11y reasons, I'd suggest using "remove" or "delete" here instead of "x". Alternatively, you could continue using an "icon" (though I think ideally we'd use an actual icon rather than the letter x), and add an `aria-label`. I think in j2html that'll probably just be something like `.withAttr("aria-label", "Remove entity")`. 

In general we need to do a big sweep to add aria-labels and alt-text to our elements - sorry for picking on this element 😄  Just noticed an easy spot to improve things now.